### PR TITLE
Upgrade to OpenEnclave v0.12.0

### LINKED ISSUE #105: Add support for Open Enclave 0.10.0
[Open Enclave 0.10.0](https://github.com/openenclave/openenclave/releases/tag/v0.10.0) [deprecates](https://github.com/openenclave/openenclave/issues/2595) the `oehostapp` CMake target in favor of `oehost`. However, simply changing our CMake targets from `oehostapp` to `oehost` causes compilation errors:

```
CMakeFiles/xgboost_enclave.dir/rabit/src/ssl_socket.cc.o: In function `rabit::utils::SSLTcpSocket::ConfigureClientSSL()':
ssl_socket.cc:(.text._ZN5rabit5utils12SSLTcpSocket18ConfigureClientSSLEv+0xbd): undefined reference to `mbedtls_debug_set_threshold'
CMakeFiles/xgboost_enclave.dir/rabit/src/ssl_socket.cc.o: In function `rabit::utils::SSLTcpSocket::ConfigureServerSSL()':
ssl_socket.cc:(.text._ZN5rabit5utils12SSLTcpSocket18ConfigureServerSSLEv+0x147): undefined reference to `mbedtls_debug_set_threshold'
/opt/openenclave/lib/openenclave/enclave/liboecore.a(cpuid.c.o): In function `oe_initialize_cpuid':
/source/enclave/core/sgx/cpuid.c:29: undefined reference to `oe_sgx_get_cpuid_table_ocall'
collect2: error: ld returned 1 exit status
enclave/CMakeFiles/xgboost_enclave.dir/build.make:960: recipe for target 'enclave/xgboost_enclave' failed
make[2]: *** [enclave/xgboost_enclave] Error 1
CMakeFiles/Makefile2:349: recipe for target 'enclave/CMakeFiles/xgboost_enclave.dir/all' failed
make[1]: *** [enclave/CMakeFiles/xgboost_enclave.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```

### PR DESCRIPTION
**Pull request by podcastinator**
Fixes #105

### CODE REVIEW

File: enclave/CMakeLists.txt [hunk 1]
```diff
@@ -56,14 +48,18 @@ if (CONSENSUS)
     list(REMOVE_ITEM CPU_SOURCES ${PROJECT_SOURCE_DIR}/src/c_api/c_api.cc)
     add_custom_command(OUTPUT xgboost_mc_t.h xgboost_mc_t.c xgboost_mc_args.h 
         DEPENDS ${PROJECT_SOURCE_DIR}/xgboost_mc.edl
-        COMMAND openenclave::oeedger8r --trusted ${PROJECT_SOURCE_DIR}/xgboost_mc.edl)
+        COMMAND openenclave::oeedger8r --trusted ${PROJECT_SOURCE_DIR}/xgboost_mc.edl
+        --search-path ${OE_INCLUDEDIR} --search-path
+        ${OE_INCLUDEDIR}/openenclave/edl/sgx)
```
Comment (chester-leung) [inline enclave/CMakeLists.txt:53]: Does this make Secure XGBoost compatible with only SGX?

**Comment by chester-leung** [comment]
I saw that you made the minimum Open Enclave version 0.12.0 in the build. Is v0.9.0 no longer compatible with Secure XGBoost after this PR?

File: enclave/CMakeLists.txt (hunk 1 above)
Comment (podcastinator) [reply enclave/CMakeLists.txt:53]: No, this just updates the path to include some SGX-specific OE .edl files (but doesn't exclude others)

**Comment by podcastinator** [comment]
Right, the build will fail will < 0.12.0. They did some refactoring, changed names of libraries etc.

**Review by chester-leung — APPROVED** [review]

**Comment by chester-leung** [comment]
Thanks!

**Pull request merged**